packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -192,65 +206,75 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
         }
     }
 
-    /* Pending new version of transfer
-    async transfer(toAddress: string, token: Token, quantity?: number, amount?: BigNumber): Promise<string> {
-        const evmToken = token as EVMToken;
+    async transfer(toAddress: string, config: TransferType): Promise<string> {
+        const evmToken = config.token;
+        if (!isEVMToken(evmToken)) {
+            throw new WalletSdkError(`Blockchain ${evmToken.chain} is not supported`);
+        }
+
         const contractAddress = evmToken.contractAddress as `0x${string}`;
         const publicClient = this.kernelClient.extend(publicActions);
         let transaction;
+        let tokenId;
         try {
-            if (amount !== undefined) {
-                // Transfer ERC20
+            if (isERC20EVMToken(evmToken)) {
                 const { request } = await publicClient.simulateContract({
                     account: this.account,
                     address: contractAddress,
                     abi: erc20,
                     functionName: "transfer",
-                    args: [toAddress, amount],
+                    args: [toAddress, (config as ERC20TransferType).amount],
```

Comment:
...then typescript should be able to infer that `config` is `ERC20TransferType` and allow this to be 
```suggestion
                    args: [toAddress, config.amount],
```
(I think)

Reply:
Sadly no, I tried that and checked again and typescript is not smart enough